hadoop/src/main/java/org/apache/iotdb/tsfile/hadoop/record/HDFSTSRecord.java

```diff
@@ -0,0 +1,204 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.tsfile.hadoop.record;
+
+import org.apache.hadoop.io.Writable;
+import org.apache.iotdb.tsfile.exception.write.UnSupportedDataTypeException;
+import org.apache.iotdb.tsfile.file.metadata.enums.TSDataType;
+import org.apache.iotdb.tsfile.utils.Binary;
+import org.apache.iotdb.tsfile.utils.StringContainer;
+import org.apache.iotdb.tsfile.write.record.TSRecord;
+import org.apache.iotdb.tsfile.write.record.datapoint.*;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author Yuan Tian
+ */
+public class HDFSTSRecord implements Writable {
+
+
+  /**
+   * timestamp of this TSRecord.
+   **/
+  private long time;
+  /**
+   * deviceId of this TSRecord.
+   **/
+  private String deviceId;
+  /**
+   * all value of this TSRecord.
+   **/
+  private List<DataPoint> dataPointList = new ArrayList<>();
+
+  public long getTime() {
+    return time;
+  }
+
+  public void setTime(long time) {
+    this.time = time;
+  }
+
+  public String getDeviceId() {
+    return deviceId;
+  }
+
+  public void setDeviceId(String deviceId) {
+    this.deviceId = deviceId;
+  }
+
+  public List<DataPoint> getDataPointList() {
+    return dataPointList;
+  }
+
+  public void setDataPointList(List<DataPoint> dataPointList) {
+    this.dataPointList = dataPointList;
+  }
+
+  public HDFSTSRecord() {
+  }
+
+  /**
+   * constructor of HDFSTSRecord.
+   *
+   * @param timestamp timestamp of this TSRecord
+   * @param deviceId  deviceId of this TSRecord
+   */
+  public HDFSTSRecord(long timestamp, String deviceId) {
+    this.time = timestamp;
+    this.deviceId = deviceId;
+  }
+
+  public TSRecord convertToTSRecord() {
+    TSRecord tsRecord = new TSRecord(time, deviceId);
+    tsRecord.dataPointList = this.dataPointList;
+    return tsRecord;
+  }
+
+  @Override
+  public void write(DataOutput out) throws IOException {
+    out.writeLong(time);
+    out.writeUTF(deviceId);
+    out.writeInt(dataPointList.size());
+    for (DataPoint dataPoint : dataPointList) {
+      out.writeShort(dataPoint.getType().serialize());
+      out.writeInt(dataPoint.getMeasurementId().getBytes(StandardCharsets.UTF_8).length);
+      out.write(dataPoint.getMeasurementId().getBytes(StandardCharsets.UTF_8));
```

Comment:
Why not just use `writeUTF()` as you have done previously?

Reply:
If there are spaces in measurementId, and I use writeUTF(), when I read, I can only use the corresponding readUTF(), I will miss the characters after the first space. This will also cause the rest data to be unusable.

Reply:
And I should change the previous deviceId's reading and writing way.